bioindex/api/bio.py

```diff
@@ -326,12 +333,45 @@ async def api_schema(req: fastapi.Request):
     return graphql.utilities.print_schema(gql_schema)
 
 
+@router.get('/bgcompress/poll/{job_id}', response_class=fastapi.responses.ORJSONResponse)
+async def api_batch_job_status(job_id: str):
+    status = aws.get_bgzip_job_status(job_id)
+    if not status:
+        raise fastapi.HTTPException(status_code=404, detail=f"Could not find job {job_id}")
+    return {
+        'status': status
+    }
+
+
+@router.post('/bgcompress/{idx}/{job_type}', response_class=fastapi.responses.ORJSONResponse)
```

Comment:
Post is a bit tough. It also feels potentially less safe since it is just public.

Could we think about moving this to main.py and make it akin to the indexing step?

```@click.command(name='index')
@click.argument('index_name')
@click.option('--rebuild', '-r', is_flag=True)
@click.option('--use-lambda', '-l', is_flag=True)
@click.option('--workers', '-w', type=int, default=None)
@click.confirmation_option(prompt='This will build the index; continue? ')
@click.pass_obj
def cli_index(cfg, index_name, use_lambda, rebuild, workers):
```

So a cli which would take the index_name ask you to confirm you want to compress the data etc. and then the thing like job type would be a flag. Does that make sense? Same would go for set-compressed.

Reply:
Yeah, I think I get your meaning.  Have a look at the updated code and let me know.